addons/info/src/index.js

```diff
@@ -82,10 +83,17 @@ function addInfo(storyFn, context, infoOptions) {
   return <Story {...props}>{storyFn(context)}</Story>;
 }
 
-export const withInfo = textOrOptions => {
-  const options = typeof textOrOptions === 'string' ? { text: textOrOptions } : textOrOptions;
-  return storyFn => context => addInfo(storyFn, context, options);
-};
+export const withInfo = makeDecorator({
+  name: 'withInfo',
+  parameterName: 'info',
+  wrapper: (getStory, context, { options, parameters }) => {
```

Comment:
Do we want to skip the addon if they don't set options or parameters? Either by passing `skipIfNoOptionsOrParameters: true` or manually doing it here..

Reply:
Def agree with the `disable: true` part, will do. I'm not so sure about the `skipIfNoOptionsOrParameters` part though. I assume the both of them do the same thing, namely disable the addon entirely. But `info` is slightly different from the others addons imo that it does something even without configuration: display Proptypes, the component, etc.

Example: https://github.com/storybooks/storybook/pull/3697/files#diff-c915ecbefba3a2734ebd93eb258431dbR353 This is basically an parameters- and options-less story, besides some minor options (header styling) and parameters (inline and manual text). The configuration don't add a lot, but the addon itself adds a lot of information stand-alone already.

So I would say just only documenting the disable flag should be the only thing

Reply:
Ok, makes sense. Thanks for thinking about that.